torch/_decomp/decompositions.py

```diff
@@ -1490,6 +1490,8 @@ def addmv(self: Tensor, mat1: Tensor, vec: Tensor, beta: int = 1, alpha: int = 1
     out = alpha * torch.mv(mat1, vec)
     if beta == 0:
         return out
+    if out.numel() == 0:  # handle empty matrix
```

Comment:
why not check `mat1.numel()`?

Reply:
I think both `out.numel()==0` and `mat1.numel()` are equivalent. Since the root cause here is to avoid `[] + non-empty tensor = []` when running `out + beta * self`, adding a checker to directly check `out.numel` seems more straightforward to me.
That said, I am open to change `out.numel()==0` to `mat1.numel()` so the condition will be consistent with the eager's implementation:https://github.com/pytorch/pytorch/blob/e97b97af56204230f1030bd297dda9bc6b053a4c/aten/src/ATen/native/Blas.cpp#L62